### PR TITLE
[DEVOPS-859] nix-darwin improvements

### PR DESCRIPTION
**Pull request by disassembler**
I'm using this locally on my home network to manage my macs and I've made some improvements I thought we might want in iohk-ops.

### CODE REVIEW

File: nix-darwin/NukeNix.hs [hunk 1]
```diff
@@ -28,6 +32,7 @@ main = sh $ do
                     , ".nixpkgs"
                     , ".config/nixpkgs"
                     ]
+  homeDir <- liftIO $ getEnv "HOME"
```
Comment (rvl) [inline nix-darwin/NukeNix.hs]: Can use `Turtle.home`

File: nix-darwin/PrepareDarwin.hs [hunk 2]
```diff
@@ -78,7 +87,11 @@ setupSSLCert = unlessM (testfile cert) $ mapM_ sudo setup
 prepareConfigs :: Shell ()
 prepareConfigs = do
   -- prepare configs for nix darwin
-  mapM_ moveAway ["/etc/bashrc", "/etc/nix/nix.conf"]
+  user <- liftIO $ getEnv "USER"
```
Comment (rvl) [inline nix-darwin/PrepareDarwin.hs]:
```
need "USER" >>= \case
    Just user -> do { ... }
    Nothing -> echo "USER not set"
```

File: nix-darwin/PrepareDarwin.hs [hunk 3]
```diff
@@ -101,9 +124,10 @@ chopProfile p = do
   sudo ["cp", tt temp, tt p]
 
 -- | This is needed so that nix-copy-closure to this host will work
-nixCopyClosureHack :: Shell ()
-nixCopyClosureHack = unlessM (testpath "/usr/bin/nix-store") $
-  sudo ["ln", "-sf", "/nix/var/nix/profiles/default/bin/nix-store", "/usr/bin"]
+setupUserBashrc :: IO ()
+setupUserBashrc = do
+  homeDir <- getEnv "HOME"
```
Comment (rvl) [inline nix-darwin/PrepareDarwin.hs]:
```
bashRC <- (</> ".bashrc") <$> home
```

Good fix -- better than symlinking `/usr/bin/nix-store` :-)

File: nix-darwin/PrepareDarwin.hs [hunk 4]
```diff
@@ -135,6 +170,8 @@ setupBuildkiteDir bk = do
     printf ("Placeholder token file: " % fp % "\n") token
     liftIO $ writeTextFile token ""
 
+
+-- | Create a directory for secret files which shouldn't go in nix store.
```
Comment (rvl) [inline nix-darwin/PrepareDarwin.hs]: This comment is out of place.

File: nix-darwin/RewriteLibs.hs [hunk 5]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env nix-shell
```
Comment (rvl) [inline nix-darwin/RewriteLibs.hs]: This module has a shebang but no main function. Maybe set this up as a `rewrite-libs` program with main and replace call to `patch-prepare` with `rewrite-libs --out-dir $out/bin $out/bin/*`

File: nix-darwin/everything.hs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+module Main where
+
+import           Data.Monoid
+import           NukeNix
+import           PrepareDarwin
+import           System.Environment
+
+main :: IO ()
+main = do
```
Comment (rvl) [inline nix-darwin/everything.hs]: Might be more direct to use LambdaCase and write
```
main = progname >>= \case
  "prepare" -> prepare
  "nuke-nix" -> nuke
  _ -> echo "wtf"
```

File: nix-darwin/PrepareDarwin.hs [hunk 7]
```diff
@@ -76,10 +85,16 @@ setupSSLCert = unlessM (testfile cert) $ mapM_ sudo setup
...
 prepareConfigs :: Shell ()
-prepareConfigs = do
+prepareConfigs =
   -- prepare configs for nix darwin
-  mapM_ moveAway ["/etc/bashrc", "/etc/nix/nix.conf"]
-  chopProfile "/etc/profile"
+  need "USER" >>= \case
+    Just user -> do
+      mapM_ moveAway ["/etc/nix/nix.conf"]
+      let contents = fromString $ "trusted-users = " <> T.unpack user
```
Comment (cleverca22) [inline nix-darwin/PrepareDarwin.hs]: if you get rid of both the `fromString` and the `T.unpack` it should remain as a `Text` and not need any conversion

File: nix-darwin/PrepareDarwin.hs [hunk 8]
```diff
@@ -135,6 +171,7 @@ setupBuildkiteDir bk = do
     printf ("Placeholder token file: " % fp % "\n") token
     liftIO $ writeTextFile token ""
 
+
```
Comment (cleverca22) [inline nix-darwin/PrepareDarwin.hs]: extra newline added

**Pull request merged**